### DISCUSSION TITLE
Webhook sending PUT request with binary data to an API

### DISCUSSION THREAD

**Discussion by jbp35** [directus/directus#17270] [Archive]
Hi, 

I would like to send a PUT request to an API with the binary data of a file stored in my library. I tried to achieve this using a webhook, but the request body seems to be limited to json content. 

Is there a way to this or any alternative solution to send binary data to an API ?

**Comment by paescuj** [comment]
To achieve this, you can create a [Custom API Hooks extension](https://docs.directus.io/extensions/hooks.html#custom-api-hooks). Possibly also via a [Flow](https://docs.directus.io/configuration/flows.html) and the help of the [Run Script operation](https://docs.directus.io/configuration/flows/operations.html#run-script).

You'll find an example of how a file stream can be retrieved over here: https://github.com/directus/directus/discussions/17149#discussioncomment-4687407
The stream could then be piped into the PUT request.

Please note that the webhooks itself will be removed in favor of flows and custom extensions in the future: https://github.com/directus/directus/issues/15553
